subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -720,6 +733,216 @@ void pe_drs_send_swap_exit(void *obj)
 	usbc_timer_stop(&pe->pd_t_sender_response);
 }
 
+/**
+ * @brief 8.3.3.4.2.1 PE_SNK_Send_Soft_Reset State
+ */
+void pe_soft_reset_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/* Reset the protocol layer */
+	prl_reset(dev);
+
+	/* Initialize PE Submachine */
+	pe->submachine = 0;
+}
+
+void pe_soft_reset_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	if (prl_is_running(dev)) {
+		switch (pe->submachine) {
```

Comment:
Can you please avoid unnecessary nesting if possible?

```c

	if (!prl_is_running(dev)) {
		return;
	}

	switch (pe->submachine) {
```

Reply:
Done